packages/sdk-core/src/modules/storage/FileSystem.ts

```diff
@@ -0,0 +1,23 @@
+import { BacktraceAttachment } from '../../model/attachment';
+
+export interface FileSystem {
+    readDir(dir: string): Promise<string[]>;
```

Comment:
quick idea:

based on the code we had some assumptions or we needed to pass some options - like for example:
- we expect to read the directory and return all files with extensions 
- we also expect to return files in specific string format - utf8

Shouldn't we note those assumptions in the interface to avoid potential surprises later? 
Previous code:

```
  const databaseFiles = fs.readdirSync(databasePath, {
            encoding: 'utf8',
            withFileTypes: true,
        });
```

Reply:
So `withFileTypes` returns not extensions, but objects instead that have `isFile` or `isDir` methods. I opted for skipping that as we will have no directories being created with the names that we search for.

Also, `utf8` is default:
![image](https://github.com/backtrace-labs/backtrace-javascript/assets/9085864/0b2ca92c-a874-44b0-9971-1a8f3023fd14)